src/transactions/slice.ts

```diff
@@ -0,0 +1,257 @@
+import { createSlice, type PayloadAction } from '@reduxjs/toolkit'
+import { REHYDRATE, type RehydrateAction } from 'redux-persist'
+import { getRehydratePayload } from 'src/redux/persist-helper'
+import {
+  TransactionStatus,
+  type EarnClaimReward,
+  type EarnDeposit,
+  type EarnSwapDeposit,
+  type EarnWithdraw,
+  type Fee,
+  type NetworkId,
+  type NftTransfer,
+  type PendingStandbyTransaction,
+  type StandbyTransaction,
+  type TokenApproval,
+  type TokenExchange,
+  type TokenTransaction,
+  type TokenTransfer,
+} from 'src/transactions/types'
+
+type BaseStandbyTransactionType<T> = Omit<PendingStandbyTransaction<T>, 'timestamp' | 'status'>
+
+export type BaseStandbyTransaction =
+  | BaseStandbyTransactionType<TokenTransfer>
+  | BaseStandbyTransactionType<TokenExchange>
+  | BaseStandbyTransactionType<TokenApproval>
+  | BaseStandbyTransactionType<NftTransfer>
+  | BaseStandbyTransactionType<EarnDeposit>
+  | BaseStandbyTransactionType<EarnSwapDeposit>
+  | BaseStandbyTransactionType<EarnWithdraw>
+  | BaseStandbyTransactionType<EarnClaimReward>
+
+// this type would ideally be TransactionReceipt from viem however the numbers
+// are of type bigint which is not serializable and causes problems at runtime
+type BaseTransactionReceipt = {
+  status: TransactionStatus
+  block: string
+  transactionHash: string
+  fees?: Fee[]
+}
+
+export type UpdateTransactionsPayload = PayloadAction<{
+  networkId: NetworkId
+  transactions: TokenTransaction[]
+}>
+
+type TransactionsByNetworkId = {
+  [networkId in NetworkId]?: TokenTransaction[]
+}
+
+interface State {
+  // Tracks transactions that have been initiated by the user
+  // before they are picked up by the chain explorer and
+  // included in the tx feed. Necessary so it shows up in the
+  // feed instantly.
+  standbyTransactions: StandbyTransaction[]
+  transactionsByNetworkId: TransactionsByNetworkId
+  feedFirstPage: TokenTransaction[]
+}
+
+const initialState: State = {
+  standbyTransactions: [],
+  transactionsByNetworkId: {},
+  feedFirstPage: [],
+}
+
+// export for testing
+export const _initialState = initialState
+
+const slice = createSlice({
+  name: 'transactions',
+  initialState,
+  reducers: {
+    addStandbyTransaction: (
+      state,
+      action: PayloadAction<{ transaction: BaseStandbyTransaction }>
```

Comment:
nit: I guess we could have done that too to avoid the arguments change
```suggestion
      action: PayloadAction<BaseStandbyTransaction>
```

No need to change now though.

Reply:
@jeanregisser I totally agree and I still prefer to change it now. This is a small change and it will provide a slight simplification of use for sure.